src/core/nvector.h

```diff
@@ -0,0 +1,94 @@
+/*
+    Copyright (C) 2018, 2021 Robert Lipe, gpsbabel.org
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+ */
+
+#ifndef NVECTOR_H
+#define NVECTOR_H
+
+#include "vector3d.h"
+
+namespace gpsbabel
+{
+
+#define COMPARE_BEARING_TO_GRTCIRC
+#ifdef COMPARE_BEARING_TO_GRTCIRC
+constexpr double MEAN_EARTH_RADIUS_METERS = 6378137.0;
+#else
+constexpr double MEAN_EARTH_RADIUS_METERS = 6371000.0;
+#endif
+constexpr double WGS84_SEMI_MAJOR_AXIS_METERS = 6378137.0; // a
+#ifdef COMPARE_BEARING_TO_GRTCIRC
+constexpr double WGS84_FLATTENING = 0.0;
+#else
+constexpr double WGS84_FLATTENING = 1.0/298.257223563; // (a-b)/a
+#endif
+constexpr double WGS84_ASPECT_RATIO = 1.0 - WGS84_FLATTENING; // b/a
+constexpr double WGS84_SEMI_MINOR_AXIS_METERS = WGS84_SEMI_MAJOR_AXIS_METERS * WGS84_ASPECT_RATIO; // b
+constexpr double WGS84_ECCENTRICITY_SQUARED = 1.0 - (WGS84_ASPECT_RATIO * WGS84_ASPECT_RATIO);
+
+class PVector;
+
+class LatLon
+{
+public:
+  LatLon(double latitude, double longitude);
+
+  double lat;
```

Comment:
Can class members be foo_ instead of foo? Or is that too mucky given how prevalent thinks like this and vec3d are in our code.

Reply:
I'd consider ditching LatLon entirely.  When I wrote test cases to compare nvector results to the exiting grtcirc.cc it wasn't very useful because the rest of our code uses independent doubles.  And these days I would probably prefer a pair even though the members are homogeneous.  structured binding declarations in c++17 make pairs/tuples even better.

I actually tried to change the Vector3D fields as you suggest but I got into a battle with clion I didn't want to fight.  I could try again.  I note our getters for that class violate https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c131-avoid-trivial-getters-and-setters, and if we wanted to heed that guideline then the trailing underscore is undesirable.  But, at this point, the fields are not public and could use a trailing underscore.

Reply:
I renamed the member variables of Vector3D as you suggested.